### PR TITLE
fix: imports

### PR DESCRIPTION
**Pull request by PythonIntermediateCoder**
## Summary
This pull request removes a useless import, It's not an issue but useless imports should not be in the code.

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by Lulalaby** [comment]
Please add a summary why you did the changes.

File: discord/commands/core.py [hunk 1]
```diff
@@ -40,8 +40,6 @@
     Optional,
     Union,
     TYPE_CHECKING,
-    Awaitable,
-    overload,
```
[the remark this answers is not in the document]
Comment (PythonIntermediateCoder) [reply discord/commands/core.py:44]: I will remove it so

[the remark this answers is not in the document]
Comment (PythonIntermediateCoder) [reply discord/commands/core.py:44]: But It would be unnecessary if I remove it, because pr #863 has an auto-merge.

**Comment by krittick** [comment]
This is an invalid PR which removes many imports that are used in the files where the changes are made.

**Pull request closed without merging**